test/multi_node_test_2.py

```diff
@@ -34,6 +43,30 @@ def test_cluster():
     assert not any(node.any_processes_alive() for node in g.list_all_nodes())
 
 
+def test_internal_config():
+    cluster = Cluster(
+        initialize_head=True,
+        connect=True,
+        head_node_args={
+            "resources": dict(CPU=1),
+            "_internal_config": json.dumps({
+                "num_heartbeats_timeout": 20
+            })
+        })
+    worker = cluster.add_node()
+    cluster.wait_for_nodes()
+
+    cluster.remove_node(worker)
+    cluster.wait_for_nodes(retries=10)
+    assert ray.global_state.cluster_resources()["CPU"] == 2
```

Comment:
This test looks like it could be flaky because of the reliance on timing..

Reply:
Hm, the test runs on Jenkins which I think has a lot less timing issues? Also, the timing leeway is on the order of seconds. 

Is there a better way to guarantee node removal other than refreshing the client table?

Alternatively, we could see if this is an issue and if we ever see it again, we could use flaky (https://github.com/box/flaky)